scipy/optimize/optimize.py

```diff
@@ -865,7 +865,7 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
     return result
 
 
-def approx_fprime(xk, f, epsilon, *args):
+def approx_fprime(xk, f, epsilon=_epsilon, *args):
```

Comment:
It won't be clear from the docstring what that value is, does it work if we put `np.sqrt(np.finfo(float).eps)` here?

Back compatibility is important. I find calling this kind of signature a little bit confusing:

```
>>> def approx_fprime(xk, f, epsilon=1.0, *args):
...     print(xk, f, epsilon, args)
...
>>> approx_fprime(1, 2, epsilon=1.0, 1)
  File "<stdin>", line 1
    approx_fprime(1, 2, epsilon=1.0, 1)
                                      ^
SyntaxError: positional argument follows keyword argument
>>> approx_fprime(1, 2, epsilon=0.5, 1)
  File "<stdin>", line 1
    approx_fprime(1, 2, epsilon=0.5, 1)
                                      ^
SyntaxError: positional argument follows keyword argument
>>> approx_fprime(1, 2, epsilon=0.5, *(1.0,))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: approx_fprime() got multiple values for argument 'epsilon'
>>> approx_fprime(1, 2, epsilon=0.5, *(1.0,2.0))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: approx_fprime() got multiple values for argument 'epsilon'
>>> approx_fprime(1, 2, 0.5, *(1.0,2.0))
1 2 0.5 (1.0, 2.0)
```

See https://stackoverflow.com/questions/58018545/why-does-python-allow-args-after-keyword-arguments

I'm not sure the user confusion will be worth the amended PR.

Reply:
Regarding documenting the default value: I went for the same wording as in check_grad.

Regarding the API: Actually this doesn't change the API; whether before or after this PR `epsilon` was effectively positional-only if you wanted to pass `args`:
```
In [1]: def f(eps, *args): print(eps, args)

In [2]: f(eps=1, 2, 3)
  File "<ipython-input-2-2c21b3b3c320>", line 1
    f(eps=1, 2, 3)
                 ^
SyntaxError: positional argument follows keyword argument


In [3]: f(eps=1, *(2, 3))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-fee0e799560d> in <module>
----> 1 f(eps=1, *(2, 3))

TypeError: f() got multiple values for argument 'eps'
```
(In other words, regardless of whether `epsilon` has a default value or not, if you want to pass `args`, then you always need to pass `epsilon` positionally, not by keyword.)
I agree the signature is actually quite bad, but again this PR doesn't change the situation (perhaps `epsilon` should instead have been keyword-only, so the signature would have been `approx_fprime(xk, f, *args, epsilon=_epsilon)`, but OTOH that would have been somewhat annoying to use when `epsilon` had no default, and is in any case a much bigger change compat-wise).